Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -768,132 +767,165 @@ namespace OpenInfraPlatform {
+					}
+				}
+				
+				void convertIfcFaceBoundList(
+					std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcFaceBound>> ifcFaceBounds,
+					const carve::math::Matrix& pos,
+					std::vector<std::vector<carve::geom::vector<3>>>& faceBoundLoops) const noexcept(true)
+				{
+					// As carve expects outer boundary of face to be at first index, outer boundary has to be moved to index 0 and inner boundary has index >= 1
+					for (auto it = ifcFaceBounds.begin(); it != ifcFaceBounds.end(); it++)
+					{
+						if (it->isOfType<typename IfcEntityTypesT::IfcFaceOuterBound>())
+						{
+							// swap element 'it' with first element
+							std::rotate(ifcFaceBounds.begin(), it, it + 1);
 
-					// Loop through all boundary definitions
-					int boundID = -1;
+							// according to ifc-documentation, only one boundary (= loop) can be outer boundary:
+							// "One loop is optionally distinguished as the outer loop of the face."
+							// (https://standards.buildingsmart.org/IFC/DEV/IFC4_3/RC3/HTML/link/ifcface.htm)
+							// thus, break the for-loop after THE outer boundary was found
+							break;
+						}
+					}
 
-					// If polygon has more than 3 vertices, then we have to project polygon into 2D, so that carve can triangulate the mesh
-					ProjectionPlane plane = UNDEFINED;
+					faceBoundLoops.resize(ifcFaceBounds.size());
+					for (size_t i = 0; i < ifcFaceBounds.size(); i++)
+					{
+						// get vertices of one bound loop into faceBoundLoops[i]
+						convertIfcFaceBound(ifcFaceBounds[i], pos, faceBoundLoops[i]);
+					}
+				}
+
+				void convertIfcFaceBound(
+					const EXPRESSReference<typename IfcEntityTypesT::IfcFaceBound>& bound,
+					const carve::math::Matrix& pos,
+					std::vector<carve::geom::vector<3>>& loopVertices3D) const noexcept(false)
+				{
+					//	IfcLoop   has subtypes   IfcEdgeLoop, IfcPolyLoop, IfcVertexLoop
+					const EXPRESSReference<typename IfcEntityTypesT::IfcLoop>& loop = bound->Bound;
+					bool polyOrientation = bound->Orientation;
+
+					if (!loop) {
+						throw oip::InconsistentModellingException(bound, " No valid loop");
+					}
 
-					// As carve expects outer boundary of face to be at first index, outer boundary has index 0 and inner boundary has index 1
-					for (const auto& bound : face->Bounds)
+					// Collect all vertices of the current loop
+					curveConverter->convertIfcLoop(loop, loopVertices3D);
+
+					for (auto& vertex : loopVertices3D)
 					{
-						if (bound.isOfType<typename IfcEntityTypesT::IfcFaceOuterBound>())
-						{
-							modBounds.insert(modBounds.begin(), bound);
-						}
-						else 
-						{
-							modBounds.push_back(bound);
-						}
+						vertex = pos * vertex;
 					}
-					modBounds.shrink_to_fit();
 
-					for (const auto& bound : modBounds) 
+					if (loopVertices3D.size() < 3)
 					{
-						boundID++;
+						throw oip::InconsistentGeometryException(loop, " Number of vertices < 3");
+					}

```

Comment:
You can test this before the for-loop applying the position.

Reply:
Done in commit 72e884d.